[임희서] Sprint2

Imgaic opened the pull request:
## 요구사항

### 기본(Sprint1)

메인 페이지
- [x] 랜딩 페이지의 url path는 루트(‘/’)로 설정합니다.
- [x] title은 “판다마켓”로 설정합니다.
- [x] 화면의 너비가 1920px 이상이면 하늘색 배경색은 너비를 꽉 채우도록 채워지고, 내부 요소들의 위치는 고정되고, 여백만 커지도록 합니다.
- [x] 화면의 너비가 1920px 보다 작아질 때, “판다마켓” 로고의 왼쪽 여백 200px“로그인" 버튼의 오른쪽 여백 200px이 유지되고, 화면의 너비가 작아질수록 두 요소간 거리가 가까워지도록 설정합니다.
- [x] 클릭으로 기능이 동작해야 하는 경우, 사용자가 클릭할 수 있는 요소임을 알 수 있도록 CSS 속성 cursor: pointer 로 설정합니다.
- [x] “판다마켓” 클릭 시 루트 페이지(‘/’)로 이동합니다.
- [x] '로그인'버튼 클릭 시 로그인 페이지(‘/login’)로 이동합니다 (빈 페이지)
- [x] “구경하러가기”버튼 클릭 시(’/items’)로 이동합니다.(빈 페이지)
- [x] “Privacy Policy”, “FAQ”는 클릭 시 각각 Privacy 페이지(‘/privacy’), FAQ 페이지(‘/faq’)로 이동합니다.(모두 빈 페이지)
- [x] 페이스북, 트위터, 유튜브, 인스타그램 아이콘을 클릭 시 각각의 홈페이지로 새로운 창이 열리면서 이동합니다.

### 심화

- [x] 사용자의 브라우저가 크고 작아짐에 따라 페이지의 요소간 간격, 요소의 크기, font-size 등 모든 크기와 관련된 값이 크고 작아지도록 설정해 보세요.(설정값은 자유입니다)

### 기본(Sprint2)

로그인 페이지, 회원가입 페이지 공통
- [x] “판다마켓" 로고 클릭 시 루트 페이지(“/”)로 이동합니다.
- [x] 로그인 페이지, 회원가입 페이지 모두 로고 위 상단 여백이 동일합니다.
- [x] input 요소에 focus in 일 때, 테두리 색상은 #3692FF입니다.
- [x] input 요소에 focus out 일 때, 테두리는 없습니다.
- [x] SNS 아이콘들은 클릭시 각각 실제 서비스 홈페이지로 이동합니다.

로그인 페이지

- [x] “회원가입”버튼 클릭 시 “/signup” 페이지로 이동합니다.

회원가입 페이지

- [x] “로그인”버튼 클릭 시 “/login” 페이지로 이동합니다.

### 심화

- [x] 비밀번호 input 요소 오른쪽에 눈모양 아이콘을 추가합니다. (추후 클릭으로 비밀번호를 보거나 가릴 수 있도록 기능을 만들어 갈 예정입니다.)
- [x] palette에 있는 color값들을 css 변수로 등록하고 사용해 주세요.

## 주요 변경사항(Sprint2)

- reset.css 사용하기 시작했습니다.
- 메인페이지에서 header 부분이 이제 위에 붙도록 수정했습니다.
- 링크가 매끄럽게 연결되도록 주소부분을 고쳐줬어요.

## 멘토에게(Sprint2)

- input에서 글자가 세로로 가운데 정렬을 하고 싶은데 안되었습니다.
뭔가 관련된 녀석은 line-height 같아요.
아이콘을 넣을 때도 이 옵션이 적용되더라고요. reset.css에 기본적으로 line-height: 1px;이게 설정되어있어서 그런거같아요.
지금은 input 창에 placeholder가 가운데에 정렬이 된 거 같기도 하고 아닌거 같기도 하고 헷갈리네요.

- reset.css를 html에서 링크했는데, 강혹 어떤 태그중에 input 같은거 보면 user agent style, 즉 크롬에서 기본적으로 제공되는 세팅이 reset.css보다 위쪽에, 더 우선순위가 높은 경우가 있다는 것을 발견했어요. 보통 유저가 작성한 것이 우선순위가 높다고 알고 있기 때문에 의문이 생기네요..

Commit:
index.html, style.css, assets폴더 및 그 안의 사진들 추가

Commit:
일단 수정사항 없이 시작

Commit:
login.html, login-register-style.css 및 에셋 추가, index.html 수정

index.html 속 로그인 링크 수정

Commit:
register.html 추가

login과register.html의 공통된 클래스 이름 변경->포괄적으로, input에focus 되었을 때 색상변경

Commit:
style.css header를 맨 위에 고정

Commit:
Merge branch 'part1-임희서-sprint2' into part1-임희서-sprint1

ding-co commented on assets/home_01.png:
assets/에 이미지가 모두 함께 같이 있는데
아이콘, 이미지 별로 폴더를 분리하면 좋을 것 같아요.
예를 들어 images/ icons/ 이렇게요~

Imgaic replied:
폴더 정리를 한번 싹 하고 분류했습니다. 그런데 그로 인해 루트( / )가 index.html이 아니게 되었습니다.
스프린트1 요구사항이었는데, 이런식으로 해도 되나 궁금합니다

ding-co replied:
참고만 해주시고 스프린트 요구사항에 맞게 해주세요~

Imgaic replied:
넵

ding-co commented on index.html:
```diff
@@ -0,0 +1,82 @@
+<!DOCTYPE html>
+<html lang="kor">
```
혹시 "ko"를 헷갈리신 것 같아요!

Imgaic replied:
수정했습니다!

ding-co commented on index.html:
```diff
@@ -0,0 +1,82 @@
+  <title>판다마켓</title>
+  <link rel="icon" href="./assets/panda-icon.png">
+  <link rel="stylesheet" href="style.css">
+  <link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/gh/orioncactus/pretendard/dist/web/static/pretendard.css" />
```
폰트 가져오는 것 너무 좋습니다.
추가적으로 폰트를 가져올 때 그냥 모든 것을 한번에 가져올 수도 있고 다이나믹 서브셋, 가변 다이나믹 서브셋, 로컬 폰트 적용 등 여러 방법이 있는데 한번 참고해보시면 좋을 것 같아요.

https://github.com/orioncactus/pretendard?tab=readme-ov-file

Imgaic replied:
한번에 가져오는 방법이 있고(현재),
필요한 유니코드의 범위에서만 사용할 만큼 가져올 수 있고 (다이나믹 서브셋),
그냥 로컬에 있는걸 가져다 쓸 수도 있고(로컬 폰트)
로 정리 했는데 가변 다이나믹 서브셋은 더 적은 용량을 쓴다는데 뭔지 잘 모르겠네요.. ㅎㅎ

ding-co replied:
네 다이나믹 서브셋으로 유니코드 범위 내에서 지정하여 한글/영어 등 다국어 구현할 때 언어별로 폰트를 셋팅할 때 사용할 수도 있습니다. 또한 사용자가 웹 페이지를 방문할 때 필요한 글자만을 포함하는 폰트의 서브셋을 실시간으로 생성하여 제공해요.
가변 다이나믹 서브셋 같은 경우에는 다이나믹 서브셋을 좀 더 발전시켜서 단일 폰트 파일에서 weight 등 변형까지도 지원하여 좀 더 빠르고 용량이 적게 제공된다고 해요.

Imgaic replied:
아하! 감사합니다. 알아둘게요.

ding-co commented on index.html:
```diff
@@ -0,0 +1,82 @@
+</head>
+<body>
+  <header>
+    <div class="logo-login">
```
css 컨벤션이나 규칙이 여러가지가 있는데 대표적으로 BEM(Block Element Modifier)가 있어요.
한번 찾아보세요~

Imgaic replied:
찾아봤는데, 아직은 100프로 공감되진 않지만, 규모가 커지고 양이 많아질수록 빛을 발할 것이라고 생각이 드네요, 3미션 부터는 한번 바꿔보겠습니다!

ding-co commented on index.html:
```diff
@@ -0,0 +1,82 @@
+  <header>
+    <div class="logo-login">
+      <a class="logo" href="/">
+        <img class="logo-image" src="./assets/logo-image.png" alt="판다마켓 로고">
```
alt를 그냥 가볍게 넘어갈 수 있는데 고려해주셔서 좋아요!

Imgaic replied:
없던 것들도 다 추가했습니다!

ding-co commented on index.html:
```diff
@@ -0,0 +1,82 @@
+  </header>
+  <main>
+    <div class="container">
+      <section id="main-top">
```
main, section 등 여러 시맨틱한 요소를 사용하시려는 것 너무 좋습니다.

Imgaic replied:
감사합니다. 의미적인 거라서 정확하게 숙지해야겠네요.

ding-co commented on index.html:
```diff
@@ -0,0 +1,82 @@
+  </header>
+  <main>
+    <div class="container">
+      <section id="main-top">
```
id를 써주셨는데 어떤 경우에 id를 쓰면 좋을지 class를 쓰면 좋을지 고민하시면 좋을 것 같아요.

Imgaic replied:
사실 유일성의 차이, 우선순위 차이 등이 있어서, 정말 유일할때는 id를 쓴다고는 하지만, 저는 정말 뭐 강조가 필요하거나 그럴때가 아니면 그냥 위에 올려주신 BEM같은 규칙따라서 class만 쓸 것 같습니다.

ding-co commented on style.css:
```diff
@@ -0,0 +1,208 @@
+  box-sizing: border-box;
+}
+
+:root {
```
변수 사용하시려는거 좋습니다!

Imgaic replied:
감사합니다. 글로벌도 따로 빼는것도 한번 준비해볼게요

ding-co commented on style.css:
```diff
@@ -0,0 +1,208 @@
+
+:root {
+  --skyblue: #3692FF;
+  --white: #ffffff;
```
#fff 로 축약해서 사용할 수 있어요~

Imgaic replied:
수정했습니다!

ding-co commented on style.css:
```diff
@@ -0,0 +1,208 @@
+}
+
+html {
+  font-family: 'Pretendard';
```
보통은 맨 앞에 주 폰트를 사용하고 만약 그 폰트가 적용되지 않았을 때 뒤에 ,로 이어서 fallback font를 사용해요. 참고해주세요~

Imgaic replied:
fallback font(예비폰트) 에 'sans-serif'를 추가했습니다!

ding-co commented on style.css:
```diff
@@ -0,0 +1,208 @@
+
+body {
+  margin: 0;
+  word-break: keep-all; /*글자가 단어단위로 잘림*/
```
body 태그에 전체 적용하는게 과연 확장성 면에서 좋을까요.
각 부분에 적용하는 것은 어떨까요.

Imgaic replied:
확장성이라는 말을 듣고 생각해보니, 사실 margin은 reset.css에 이미 있고, word-break 같은 경우도 사실 모든 글에 적용이 되었으면 해서 한거긴 한데요.
"CSS를 작성할 땐 한 군데만 바꿔도 그 변화를 여러 곳에 적용시킬 수 있다. 이론적으로 이런 기능은 굉장히 매력적이다. 그러나 쓰다 보니 내가 이것을 달가워하지 않는다는 것을 발견했다." 라고 어떤 글에 써있네요. 하나의 요소만 바꾸길 바라지, 다른 것에도 영향을 주는 상황 자체를 꺼려하게 된다는 느낌이였어요. 그런 면에서 봤을 때, 저런 상위 태그에 넣는 것보다는 필요한 곳에만 넣거나, 저 word-break만을 위한 클래스 규칙을 하나 만들어서 html에 그 클래스를 필요한 곳에 추가하는 방식이 좀 더 나을 것 같습니다.
그래서 위 내용을 바탕으로 코드를 좀 바꿔 주었습니다.

Imgaic replied:
word-break 뿐 아니라, 어떤 설정이든 너무 상위 태그에 달아 놓으면, 의도치 않은 상황이 나올 수도 있다는 생각을 하게 되었습니다. 확장성을 고려한다는게 이런 사고방식인거 같아요. 미래를 고려해서 최대한 의도치 않은 상황이 나오지 않게 하는 느낌(?)

ding-co replied:
네 맞습니다~ 여러가지 레퍼런스를 찾아보면서 학습하시고 또 생각하시는게 너무 좋은 것 같네요 ㅎㅎ

ding-co commented on style.css:
```diff
@@ -0,0 +1,208 @@
+  justify-content: space-between;
+  align-items: center;
+  padding: 9.5px 200px;
+  flex-shrink: 0;
```
flex-shrink를 의도하고 사용하신지 궁금해요

Imgaic replied:
이것저것 수정하다가 그런 것 같습니다. 부모요소중에 flex인 부모가 없는데.. 쓸모없는 부분이네요. 삭제했습니다.

Imgaic replied:
찾아보다가, 왜 logo에는 없고 login에만 shrink가 있나 하고 실헝해보니까, a태그랑 다르게 img 태그는 width height를 정해주면 거기서 크기가 고정되네요. 그래서 굳이 shrink가 안필요했던 것 같습니다.

ding-co commented on index.html:
```diff
@@ -0,0 +1,82 @@
+    <div class="container">
+      <section id="main-top">
+        <div>
+          <h1 class="head description">일상의 모든 물건을<br>거래해 보세요</h1>
```
br태그는 self closing으로 `<br/> `이렇게 사용할 수도 있을 것 같아요

Imgaic replied:
이렇게 라는게 무슨 말씀이신가요? 혹시 코드 상에서 '거래해 보세요'부터를 가독성을 위해 다음줄로 넘겨도 좋다는 말씀인가요?

ding-co replied:
코드가 짤렸네요 ㅋㅋ 다시 봐주세요~

Imgaic replied:
아하 넵 ㅋㅋㅋㅋㅋ
어떤 환경에서는 닫아주는게 규칙인 곳도 있는 듯 하네요.
그리고 그냥 궁금해서 찾아보니 `<br />`이나`<br/>`이나 상관없는 것 같습니다.
기억해두겠습니다.

ding-co commented on index.html:
```diff
@@ -0,0 +1,82 @@
+        </div>
+      </section>
+    
+      <figure>
```
figure 태그 사용하신 시도 너무 좋습니다.
참고로 반응형 이미지를 위해 picture 태그를 사용하는데 참고만 해주세요~

Imgaic replied:
알겠습니다, 찾아봤는데 source태그랑 img를 하위태그로 사용해서 막 범위를 지정해서 다른 사진을 가져올 수 있게 하는 그런 느낌이네요, 기억해두겠습니다.

ding-co commented on assets/ic_google.png:
보통 아이콘은 사실 투명도를 처리할 수 있는 png보다는 svg를 사용해요.
이미지 확장자에는 png, jpg, jpeg, svg 등 여러 형식이 있는데 언제 어느때 어떤 확장자를 사용하면 좋을지 고민해보시면 좋을 것 같아요!

Imgaic replied:
png : 투명 처리 가능 (용량 높음)
jpg = jpeg : 배경 하얀색임(png보다 용량 작음)
svg : 용량 작아서 아이콘 같은거 쓸 때 적합, 확대해도 안깨짐, 복잡한 거 하면 컴퓨터에 부하 있음

정리해보았습니다. 감사해요

Imgaic replied:
ic들을 전부 다 svg로 바꿔보았는데요, 훨씬 aliasing이 심해지고 심지어는 용량도 더 높아서, 일단 다시 png로 바꿨습니다. 그래도 svg는 확대했을 때는 확실히 낫더군요.

ding-co replied:
마지막으로 애플 레티나 디스플레이 고려해서 피그마에서 이미지 다운로드 할 때 x2 배로 다운로드 받아서 쓰는 경우도 많아요.
해상도 이미지 관련 이슈인데 한번 찾아보시고 참고해주세요~

Imgaic replied:
넵, 안그래도 1배는 너무 깨진다 해야하나 해서 그냥 3배로 쓰고 있었거든요. 그런 이슈가 있었군요

ding-co commented on login-register-style.css:
```diff
@@ -0,0 +1,146 @@
+  box-sizing: border-box;
+}
+
+:root {
```
변수 분리 너무 좋습니다.
하지만 login-register-style.css에 한정시키기 보다는 외부에 global.css를 하나 만들어서 전역적으로 공통 사용할 수 있게 만들면 좋을 것 같아요!

Imgaic replied:
반영했습니다!

ding-co commented on login-register-style.css:
```diff
@@ -0,0 +1,146 @@
+  font-family: 'Pretendard';
+}
+
+body {
```
마찬가지고 html, body 등 공통 속성도 외부로 빼서 사용하면 좋을 것 같아요.

Imgaic replied:
반영했습니다!

ding-co commented on login.html:
```diff
@@ -0,0 +1,42 @@
+  </header>
+  <main>
+    <form>
+      <label for="eamil">이메일</label>
```
label의 for 속성이 input의 id 속성과 매칭되는 점을 고려해주신 것 너무 좋습니다.
참고로 label 태그 안에 input을 자식으로 넣어서 for, id 속성 사용 없이도 사용가능하다는 점 참고해주세요.
물론 스타일링이나 태그가 자식으로 들어가기 애매한 경우에는 for, id를 매칭시키는 것이 좋을거에요.

Imgaic replied:
넵, 깔끔할 것 같아서 label안에 넣어봤는데, 결과값이 다르더라구요. 굳이 바꿀 필요가 없다고 생각해서 굳이 바꾸지는 않았습니다.

ding-co commented on reset.css:
```diff
@@ -0,0 +1,48 @@
+/* http://meyerweb.com/eric/tools/css/reset/ 
```
reset css 적용 좋습니다!

Imgaic replied:
감사합니다

ding-co commented on login-register-style.css:
```diff
@@ -70,6 +70,12 @@ form label {
   width: 100%;
 }
 
+.input:focus {
```
네 그래서 보통 reset 이나 global에 input에 관련된 기본 border 등 스타일을 초기화시켜두고
outline으로 스타일을 수정하기도 해요.

Imgaic replied:
반영했습니다!

ding-co commented on style.css:
```diff
@@ -29,6 +29,9 @@ header {
   border-bottom: 1px solid #dfdfdf;
   height: 70px;
+
+  position: sticky;
```
sticky를 사용해 스크롤 되더라도 맨 위에 고정시키려는 의도 맞으실까요?

Imgaic replied:
맞습니다! 스프린트 2 요구사항에 있었기에, 그리고 간단하게 두 줄만 추가하면 되었어서 바로 넣었습니다. fixed 보다 쓰기 훨씬 편한게, sticky를 쓰면 자리를 그대로 차지하다 보니까 따로 후처리를 할게 없더라구요.

ding-co commented:
전반적으로 미션에 대한 모든 요구사항을 다 하시려고 노력하신게 많이 느껴졌고,
코드에서도 고민하신 부분과 최대한 시맨틱한 태그 등 사용하시려는 모습이 많이 느껴졌어요.
고생 많으셨어요~

ding-co commented:
```
input에서 글자가 세로로 가운데 정렬을 하고 싶은데 안되었습니다.
뭔가 관련된 녀석은 line-height 같아요.
아이콘을 넣을 때도 이 옵션이 적용되더라고요. reset.css에 기본적으로 line-height: 1px;이게 설정되어있어서 그런거같아요.
지금은 input 창에 placeholder가 가운데에 정렬이 된 거 같기도 하고 아닌거 같기도 하고 헷갈리네요.
```
line-height는 줄의 높이를 조정할 수 있는데 default로 normal 즉 1.2 입니다.
이 부분을 조정하고 싶으시면 input 관련된 스타일에서 수정하셔도 되고 별도로 커스터마이즈하게 reset.css를 수정하셔도 됩니다.
보통 padding도 같이 이용해서 세로 높이를 조절할텐데 한번 여러 시도를 해보세요.

`reset.css를 html에서 링크했는데, 강혹 어떤 태그중에 input 같은거 보면 user agent style, 즉 크롬에서 기본적으로 제공되는 세팅이 reset.css보다 위쪽에, 더 우선순위가 높은 경우가 있다는 것을 발견했어요. 보통 유저가 작성한 것이 우선순위가 높다고 알고 있기 때문에 의문이 생기네요..`
user agent style 즉 브라우저에서 기본적으로 제공되는 세팅이라 그걸 방지하기 위해 저희가 reset.css 또는 normalize 같은 것을 이용해요.
때로는 만약 말씀하신 것처럼 더 우선순위가 높은 것이 있다면 그때 트러블 슈팅을 하면서 해결해보시는게 좋을 것 같아요.
reset에 미리 선언도 해보고 안되면 해당 태그 스타일로 수정도 해보고 !important로 시도해보고 여러 방법을 통해서요.

Commit:
lang=kor -> ko 오타 수정, 모든 img alt 추가

Commit:
파일을 세 가지 폴더로 분류하고 이미지들을 두 가지 폴더로 더 분류함

원래 index.html이 루트 페이지였는데, 이로 인해 바뀌게 되었음,
스프린트1의 요구사항이었는데, 그렇다면 index.html만 밖에 꺼내놓을지 고민됨

Commit:
index.html 을 랜딩페이지, 루트(/)로 설정, fallback font 추가

Commit:
index.html에도 reset.css 적용, 확장성을 고려해 body 규칙을 지우고, 필요한 곳에만 word-break 속성 적용

Commit:
global.css 파일을 추가하고, 상위 태그 규칙과 css변수 선언, css 파일들 속 색상헥스 코드를가능한 만큼 변수로 변경

Commit:
global.css에 input의 기본 테두리를 없애는 코드 추가, 그에 따라 login-register-style.css도 outline의 색만 바꾸는 것이 아닌, 생성하도록 변경

굳이 예쁘게 설정되어 있던 굵기를 초기화해야하나 싶기도 하지만, 미래를 생각했을 때, 선택권을 받는 편이 합리적이지 않나 하고 생각함

Imgaic commented:
와 진짜 배신감 느끼는데요,

> reset.css를 html에서 링크했는데, 강혹 어떤 태그중에 input 같은거 보면 user agent style, 즉 크롬에서 기본적으로 제공되는 세팅이 reset.css보다 위쪽에, 더 우선순위가 높은 경우가 있다는 것을 발견했어요. 보통 유저가 작성한 것이 우선순위가 높다고 알고 있기 때문에 의문이 생기네요..

이거 아무리 생각해도 이상해서 계속 찾다보니까 form 태그는 폰트 속성을 상속 안한다나... 그래서 input 하고 button은 따로 해줘야 한다고.. 지금까지 폰트도 적용이 안되고 있었네요.

또 확인해보니 line-height도 상속을 안받네요 예네들은...

단순히 css stylesheet의 우선순위 문제는 아니었던 것 같습니다.

Commit:
font-family, line-height 등이 button, input 등에 적용되지 않던 것을 해결

form 관련 태그들은 폰트관련 속성들을 상속받지 않는 모양이다.

ding-co commented:
> 와 진짜 배신감 느끼는데요,
> 
> > reset.css를 html에서 링크했는데, 강혹 어떤 태그중에 input 같은거 보면 user agent style, 즉 크롬에서 기본적으로 제공되는 세팅이 reset.css보다 위쪽에, 더 우선순위가 높은 경우가 있다는 것을 발견했어요. 보통 유저가 작성한 것이 우선순위가 높다고 알고 있기 때문에 의문이 생기네요..
> 
> 이거 아무리 생각해도 이상해서 계속 찾다보니까 form 태그는 폰트 속성을 상속 안한다나... 그래서 input 하고 button은 따로 해줘야 한다고.. 지금까지 폰트도 적용이 안되고 있었네요.
> 
> 또 확인해보니 line-height도 상속을 안받네요 예네들은...
> 
> 단순히 css stylesheet의 우선순위 문제는 아니었던 것 같습니다.

네 그래서 보통 폰트 스타일이나 그런 것들은 다 사용할 때 태그에 작성해줘요
